### DISCUSSION TITLE
how to contribute with docker image for arm platforms?

### DISCUSSION THREAD

**Discussion by lumpsoid** [HugoFara/lwt#141] [Ideas]
Currently the repository only has a docker image for amd platforms, and it cannot be run on arm platforms such as raspberry pi. I checked that the dockerfile uses an image that supports arm, so there is no problem building such an image and I successfully built it for myself, but as I understand it would be more convenient if this image was already available.

Could you please tell me how I can send/upload this arm docker image as part of this repository?

**Comment by HugoFara** [comment]
Hello, thank you for your post!

What did you change to build and run a successful image for arm platforms? The images are built automatically, so the workflow would be more interesting for me than the final result :smile:

**Reply by lumpsoid** [reply]
I just have raspberry pi, and I just built image on it, but I found out that there is `buildx` plugin for docker cli, that allows to build docker images for multiple cpu architectures. 
as far as I understand you need something like this `docker buildx build -t docker.io/xx/myimage:v1 -f [path to Dockerfile] --push --platform=linux/arm64,linux/amd64 [path to build context]`

https://github.com/docker/buildx

**Reply by lumpsoid** [reply]
find out how to setup github actions to build for multiple platforms
https://docs.docker.com/build/ci/github-actions/multi-platform/

in your `docker-image.yml` in last section `with` you need to add list of platforms to build
`platforms: linux/amd64,linux/arm64`

**Comment by HugoFara** [comment]
Ouch, I was no longer working on LWT when I received your answer, and I totally forgot this thread, sorry :pray: 

Your answer was good one, with the great help of @ProgramComputer we rewrote the GitHub action to build multiple images (#169). 

Enjoy!
